src/snapred/ui/workflow/NormalizationCalibrationWorkflow.py

```diff
@@ -190,31 +196,61 @@ def callNormalizationCalibration(self, groupingFile, smoothingParameter):
 
         self._specifyNormalizationView.updateWorkspaces(focusWorkspace, smoothWorkspace)
 
-    def onNormalizationValueChange(self, index, smoothingValue):  # noqa: ARG002
+    def applySmoothingUpdate(self, index, smoothingValue, dMin):
+        from mantid.simpleapi import DeleteWorkspace
+
+        workspaces = self.responses[-1].data
+        DeleteWorkspace(Workspace=workspaces["smoothedOutput"])
+        payload = SmoothDataExcludingPeaksRequest(
+            inputWorkspace=workspaces["outputWorkspace"],
+            outputWorkspace=workspaces["smoothedOutput"],
+            samplePath=self.samplePaths[self.sampleIndex],
+            groupingPath=self.groupingFiles[index],
+            runNumber=self.runNumber,
+            smoothingParameter=smoothingValue,
+            dMin=dMin,
+        )
+        request = SNAPRequest(path="normalization/smooth", payload=payload.json())
+        response = self.interfaceController.executeRequest(request)
+        focusWorkspace = workspaces["outputWorkspace"]
+        smoothWorkspace = response.data
+        self.responses[-1].data["smoothedOutput"] = response.data
+        self._specifyNormalizationView.updateWorkspaces(focusWorkspace, smoothWorkspace)
+
+    def onNormalizationValueChange(self, index, smoothingValue, dMin):  # noqa: ARG002
         if not self.initializationComplete:
             return
+        # disable recalculate button
+        self._specifyNormalizationView.disableRecalculateButton()
 
         self.lastGroupingFile = self.groupingFiles[index]
         self.lastSmoothingParameter = smoothingValue
+        self.lastDMin = dMin
 
         groupingFileChanged = self.groupingFiles[self.initGroupingIndex] != self.lastGroupingFile
         smoothingValueChanged = self.initSmoothingParameter != self.lastSmoothingParameter
-
-        if groupingFileChanged or smoothingValueChanged:
+        dMinValueChanged = self.initDMin != self.lastDMin
+        if groupingFileChanged:
             from mantid.simpleapi import DeleteWorkspace
 
-            DeleteWorkspace(Workspace="focussedRawVanadium")
-            DeleteWorkspace(Workspace="smoothedOutput")
+            workspaces = self.responses[-1].data
+
+            # BAD! >:C This shouldnt be here. This should be in the backend
+            DeleteWorkspace(Workspace=workspaces["outputWorkspace"])
+            DeleteWorkspace(Workspace=workspaces["smoothedOutput"])
             self.initGroupingIndex = index
             self.initSmoothingParameter = smoothingValue
-            self.callNormalizationCalibration(self.groupingFiles[index], smoothingValue)
+            self.callNormalizationCalibration(self.groupingFiles[index], smoothingValue, dMin)
+        elif smoothingValueChanged or dMinValueChanged:
+            self.applySmoothingUpdate(index, smoothingValue, dMin)
```

Comment:
Should the below lines 246 and 247 be combined into one `elif`?

Reply:
yes that should work